session/sshforward/sshprovider/raw_provider.go

```diff
@@ -0,0 +1,79 @@
+package sshprovider
+
+import (
+	"context"
+	"net"
+
+	"github.com/moby/buildkit/session"
+	"github.com/moby/buildkit/session/sshforward"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+type RawConfig struct {
+	ID     string
+	Dialer func(context.Context) (net.Conn, error)
+}
+
+func NewRawProvider(confs []RawConfig) (session.Attachable, error) {
```

Comment:
Do we need to make this public? I think `Raw` should remain a property for the SSH provider.

Reply:
Fair enough, I've made this private.

Reply:
Waiting for CI before I push up changes.